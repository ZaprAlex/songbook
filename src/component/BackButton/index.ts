export { default } from './BackButton';
export * from './types';
