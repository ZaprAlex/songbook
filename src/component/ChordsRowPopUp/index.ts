import ChordsRowPopUp from './ChordsRowPopUp';

export default ChordsRowPopUp;
