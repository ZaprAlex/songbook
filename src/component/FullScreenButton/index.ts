export { default } from './FullScreenButton';
export * from './types';
