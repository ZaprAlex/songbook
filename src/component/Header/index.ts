export { default } from './Header';
export * from './types';
