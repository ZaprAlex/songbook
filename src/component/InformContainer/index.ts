export { default } from './InformContainer';
export * from './types';
