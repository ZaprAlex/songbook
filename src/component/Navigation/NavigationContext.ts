import React, { useContext } from 'react';

import { INavigation } from '../../utils/navigation/types';

/**
 * Navigation context
 */
export const AppNavigationCtx = React.createContext<INavigation | undefined>(undefined);

const useAppNavigation = () => {
    const appNavigation = useContext(AppNavigationCtx);

    if (typeof appNavigation === 'undefined') {
        throw Error(`Use useAppNavigation() inside <NavigationProvider>`);
    }

    return appNavigation;
};

export default useAppNavigation;
