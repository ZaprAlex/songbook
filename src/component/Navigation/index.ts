import NavigationProvider from './NavigationProvider';
import useAppNavigation, { AppNavigationCtx } from './NavigationContext';

export { NavigationProvider, AppNavigationCtx, useAppNavigation };
