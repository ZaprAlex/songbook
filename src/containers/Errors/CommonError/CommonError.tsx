import React from 'react';

import { useAppNavigation } from '../../../component/Navigation';
import { IButton } from '../../../component/Button';
import InformContainer from '../../../component/InformContainer/InformContainer';

import styles from '../Errors.module.css';

const CommonError: React.FC = () => {
    const { goBack } = useAppNavigation();
    const buttonProps: IButton = { size: 'x-large' };

    return (
        <div className={styles.page}>
            <InformContainer
                header="Что-то пошло не так"
                buttonLabel="Вернуться"
                onClick={() => goBack()}
                contentClassName={styles.content}
                buttonClassName={styles.button}
                buttonProps={buttonProps}
            />
        </div>
    );
};

export default CommonError;
