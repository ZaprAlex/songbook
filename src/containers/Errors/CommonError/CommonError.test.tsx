import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import { useAppNavigation } from '../../../component/Navigation';
import CommonError from './CommonError';

jest.mock('../../../component/Navigation');

describe('CommonError', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        jest.restoreAllMocks();
    });

    const goBack = jest.fn();

    (useAppNavigation as jest.Mock).mockReturnValue({ goBack });

    test('should render', async () => {
        const { container } = render(<CommonError />);

        expect(container.firstChild).toMatchSnapshot();

        fireEvent.click(screen.getByTestId('button-submit'));

        expect(goBack).toHaveBeenCalled();
    });
});
