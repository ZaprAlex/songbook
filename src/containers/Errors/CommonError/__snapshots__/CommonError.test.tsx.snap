// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonError should render 1`] = `
<div
  class="page"
>
  <div
    class="content content"
  >
    <div
      class="header"
    >
      Что-то пошло не так
    </div>
    <div
      class="buttonBlock"
    >
      <button
        class="btn btn-green btn-contained btn-x-large btnSkeuomorph button"
        data-test-id="button-submit"
        type="button"
      >
        <span
          class="buttonLabel"
        >
          Вернуться
        </span>
      </button>
    </div>
  </div>
</div>
`;
